crawler/regelleistung.py

```diff
@@ -0,0 +1,297 @@
+    sql = f"SELECT date_from FROM regelleistung.{table_name} ORDER BY date_from DESC LIMIT 1"
+    return get_date_from_sql(engine, table_name, sql)
+    
+def get_earliest_date_if_table_exists(engine, table_name):
+    sql = f"SELECT date_from FROM regelleistung.{table_name} ORDER BY date_from ASC LIMIT 1"
+    return get_date_from_sql(engine, table_name, sql)
+
+def database_friendly(string):
+    return (string.lower()
+            .replace("(eur/mw)/h", "eur_mwh")
+            .replace("productname", "product")
+            .replace("[", "")
+            .replace("]", "")
+            .replace("/", "_")
+            .replace("(", "")
+            .replace(")", "")
+            .replace("-", "")
+            .replace("+", "")
+            .replace(" ", "_")
+            # Table fcr_ergebnisse
+            .replace("fr_demand_mw", "france_demand_mw")
+            .replace("dk_demand_mw", "denmark_demand_mw")
+            .replace("nl_demand_mw", "netherlands_demand_mw")
+            .replace("at_demand_mw", "austria_demand_mw")
+            .replace("be_demand_mw", "belgium_demand_mw")
+            .replace("de_demand_mw", "germany_demand_mw")
+            .replace("ch_demand_mw", "switzerland_demand_mw")
+            .replace("si_demand_mw", "slovenia_demand_mw")
+            .replace("at_import_export_mw", "austria_deficit_surplus_mw")
+            .replace("fr_import_export_mw", "france_deficit_surplus_mw")
+            .replace("dk_import_export_mw", "denmark_deficit_surplus_mw")
+            .replace("ch_import_export_mw", "switzerland_deficit_surplus_mw")
+            .replace("si_import_export_mw", "slovenia_deficit_surplus_mw")
+            .replace("be_import_export_mw", "belgium_deficit_surplus_mw")
+            .replace("de_import_export_mw", "germany_deficit_surplus_mw")
+            .replace("nl_import_export_mw", "netherlands_deficit_surplus_mw")
+            .replace("at_settlementcapacity_price_eur_mw", "austria_settlementcapacity_price_eur_mw")
+            .replace("ch_settlementcapacity_price_eur_mw", "switzerland_settlementcapacity_price_eur_mw")
+            .replace("de_settlementcapacity_price_eur_mw", "germany_settlementcapacity_price_eur_mw")
+            .replace("si_settlementcapacity_price_eur_mw", "slovenia_settlementcapacity_price_eur_mw")
+            .replace("be_settlementcapacity_price_eur_mw", "belgium_settlementcapacity_price_eur_mw")
+            .replace("dk_settlementcapacity_price_eur_mw", "denmark_settlementcapacity_price_eur_mw")
+            .replace("nl_settlementcapacity_price_eur_mw", "netherlands_settlementcapacity_price_eur_mw")
+            .replace("fr_settlementcapacity_price_eur_mw", "france_settlementcapacity_price_eur_mw")
+        )
+
+def get_df_for_date(url, date_to_get):
+    date_str = date_to_get.strftime('%Y-%m-%d')
+    url_with_date = url.format(date_str=date_str)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
```

Comment:
which warning is ignored here..?

Reply:
This was about the style that is applied to read in the excel file and it's just a hint, that the default style is applied.

Reply:
As far as I understand is, there is no easy fix from the code side, because this hint seems to occur if the sheet name is not the default name from openpyxl. All references I found also silenced this warning. This warning doesn't occur with calamine as an engine, but I didn't wanted to add another requirement, because openpyxl is already in the requirements, but calamine not. Or would you rather add the requirement and have no warning silenced?

Reply:
Sounds good and i would not switch to calamine.

I would prefer to have something like:

```
    warnings.filterwarnings(
        action="ignore", category=MyWarning, message="This is a warning"
    )
```
instead of the global filter - making it obvious why this line was added :)
But this is just a minor improvement :)